src/cyclebane/graph.py

```diff
@@ -451,20 +380,19 @@ def to_networkx(self) -> nx.DiGraph:
         new_names = {
             node: NodeName(node.name.name, node.index)
             for node in graph
-            if isinstance(node, NodeName)
+            if isinstance(node, NodeName) and isinstance(node.name, MappedNode)
         }
         graph = nx.relabel_nodes(graph, new_names)
 
         # Get values using previously stored index values
-        for values in self._node_values.values():
-            for name, col in values.items():
-                for match in [
-                    node
-                    for node in graph.nodes
-                    if isinstance(node, NodeName) and node.name == name
-                ]:
-                    value = _get_value_at_index(col, match.index.to_tuple())
-                    graph.nodes[match][self.value_attr] = value
+        # TODO We could and should(?) probably use positional indexing here, which would
+        # be faster?
+        for node in graph.nodes:
+            if (
+                isinstance(node, NodeName)
+                and (node_values := self._node_values.get(node.name)) is not None
+            ):
+                graph.nodes[node][value_attr] = node_values.sel(node.index.to_tuple())
```

Comment:
I cannot really remember why I initially used label-based indexing here. Positional indexing makes more sense, I think I will try to change.

Reply:
Did you try with positional indexing?

Reply:
I had a look at it and it was not so simple. Maybe we can leave it for now, and think of an altogether better way of implementing this function in the future?